aten/src/TH/THGeneral.h.in

```diff
@@ -42,6 +42,12 @@
 # define TH_UNREACHABLE __builtin_unreachable();
 #endif
 
+#if (__GNUC__ > 2) || (__GNUC__ == 2 && __GNUC_MINOR__ > 4)
```

Comment:
#if defined(__GNUC__) && ((__GNUC__ > 2) || (__GNUC__ == 2 && __GNUC_MINOR__ > 4))

Reply:
`__GNUC__` evaluates to 0 if not defined. But yes I agree your change makes it clearer. I'm updating it.